### PR TITLE
fix: `ResponsiveStyleValue` not support field omit

### PR DESCRIPTION
**Pull request by kkocdko**

### CODE REVIEW

**Comment by kkocdko** [comment]
## Reproduction

```typescript
let a = <Grid container columns={{ xs: 4, sm: 8, md: 12 }} />
```

Caused `ts(2769) No overload matches this call.` on `suid`, but no problems on `mui`.

The code on mui here, it just use `key: string`.

https://github.com/mui/material-ui/blob/v5.10.6/packages/mui-system/src/index.d.ts#L95-L101

File: packages/system/src/styleFunctionSx/index.ts [hunk 1]
```diff
@@ -3,4 +3,4 @@ import { Breakpoint } from "../createTheme/createBreakpoints";
 export type ResponsiveStyleValue<T> =
   | T
   | Array<T | undefined>
-  | { [key in Breakpoint]: T | undefined };
+  | { [key in Breakpoint]?: T | undefined };
```
Comment (juanrgm) [inline packages/system/src/styleFunctionSx/index.ts:6]:
```suggestion
  | { [key in Breakpoint]?: T };
```

**Comment by juanrgm** [comment]
There is also support for custom breakpoints.

```ts
declare module "@suid/system/createTheme/createBreakpoints" {
  interface BreakpointOverrides {
    xs: false; // removes the `xs` breakpoint
    sm: false;
    md: false;
    lg: false;
    xl: false;
    mobile: true; // adds the `mobile` breakpoint
    tablet: true;
    laptop: true;
    desktop: true;
  }
}
```

Following the MUI docs, maybe I must move `BreakpointOverrides` to `@suid/material/styles`.

**Comment by juanrgm** [comment]
Thanks for the merge request.

**Pull request merged**